api/server/sdk/credentials.go

```diff
@@ -589,3 +589,194 @@ func validateAndDeleteIfInvalid(ctx context.Context, s *CredentialServer, uuid s
+	req *api.SdkCredentialCreateRequest,
+	aws *api.SdkAwsCredentialRequest,
+	credId string,
+) (*api.SdkCredentialUpdateResponse, error) {
+
+	params := make(map[string]string)
+
+	params[api.OptCredType] = "s3"
+	params[api.OptCredName] = req.GetName()
+	// Users dont have to provide there if they dont want to change this
+	if len(req.GetEncryptionKey()) != 0 {
+		params[api.OptCredEncrKey] = req.GetEncryptionKey()
+	}
+	if len(req.GetBucket()) != 0 {
+		params[api.OptCredBucket] = req.GetBucket()
+	}
+	if len(aws.GetRegion()) != 0 {
+		params[api.OptCredRegion] = aws.GetRegion()
+	}
+	if len(aws.GetEndpoint()) != 0 {
+		params[api.OptCredEndpoint] = aws.GetEndpoint()
+	}
+	if len(aws.GetAccessKey()) != 0 {
+		params[api.OptCredAccessKey] = aws.GetAccessKey()
+	}
+	if len(aws.GetSecretKey()) != 0 {
+		params[api.OptCredSecretKey] = aws.GetSecretKey()
+	}
+	// Users have to provide correct values, whether they want to change it or not
+	params[api.OptCredDisableSSL] = fmt.Sprintf("%v", aws.GetDisableSsl())
+
+	params[api.OptCredDisablePathStyle] = fmt.Sprintf("%v", aws.GetDisablePathStyle())
+	params[api.OptCredProxy] = fmt.Sprintf("%v", req.GetUseProxy())
+	params[api.OptCredIAMPolicy] = fmt.Sprintf("%v", req.GetIamPolicy())
+	params[api.OptCredStorageClass] = fmt.Sprintf("%v", req.GetS3StorageClass())
+	err := s.update(ctx, req, params, credId)
+
+	if err != nil {
+		return nil, status.Errorf(
+			codes.Internal,
+			"failed to create aws credentials: %v",
+			err.Error())
+	}
+
+	return &api.SdkCredentialUpdateResponse{}, nil
+}
+
+func (s *CredentialServer) azureUpdate(
+	ctx context.Context,
+	req *api.SdkCredentialCreateRequest,
+	azure *api.SdkAzureCredentialRequest,
+	credId string,
+) (*api.SdkCredentialUpdateResponse, error) {
+
+	params := make(map[string]string)
+
+	params[api.OptCredType] = "azure"
+	params[api.OptCredName] = req.GetName()
+	if len(req.GetEncryptionKey()) != 0 {
+		params[api.OptCredEncrKey] = req.GetEncryptionKey()
+	}
+	if len(req.GetBucket()) != 0 {
+		params[api.OptCredBucket] = req.GetBucket()
+	}
+	if len(azure.GetAccountKey()) != 0 {
+		params[api.OptCredAzureAccountKey] = azure.GetAccountKey()
+	}
+	if len(azure.GetAccountName()) != 0 {
+		params[api.OptCredAzureAccountName] = azure.GetAccountName()
+	}
+
+	params[api.OptCredProxy] = fmt.Sprintf("%v", req.GetUseProxy())
+	params[api.OptCredIAMPolicy] = fmt.Sprintf("%v", req.GetIamPolicy())
+
+	err := s.update(ctx, req, params, credId)
+
+	if err != nil {
+		return nil, status.Errorf(
+			codes.Internal,
+			"failed to update Azure credentials: %v",
+			err.Error())
+	}
+
+	return &api.SdkCredentialUpdateResponse{}, nil
+}
+
+func (s *CredentialServer) googleUpdate(
+	ctx context.Context,
+	req *api.SdkCredentialCreateRequest,
+	google *api.SdkGoogleCredentialRequest,
+	credId string,
+) (*api.SdkCredentialUpdateResponse, error) {
+
+	if len(google.GetJsonKey()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply JSON Key")
```

Comment:
Why are we not doing the same checks for azure and aws ?

Reply:
Thanks!, actually these checks are not needed. Users need not provide these, if they dont want to modify them. I should take these checks out.